### PR TITLE
Update sc

### PR DESCRIPTION
**Pull request by gadfort**

### CODE REVIEW

**Review by nmoroze — CHANGES_REQUESTED** [review]
Awesome cleanup, thanks so much for taking care of this! I left a couple comments, mostly just style nits/typos.

I also tested the changes locally and successfully ran a two-stage build.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
```
[the remark this answers is not in the document]
Comment (gadfort) [reply .flake8:1]: yes

File: floorplan.py [hunk 2]
```diff
@@ -424,14 +121,12 @@ def top_floorplan(fp):
                 pin_name = 'vddio'
             elif pad_type == VSSIO:
                 pin_name = 'vssio'
-            pad_name = f'{pin_name}{i}'
+            pad_name = f'padring.we_pads\\[0\\].i0.pad{pin_name}\\[0\\].i0.io{pin_name}'
```
[the remark this answers is not in the document]
Comment (gadfort) [reply floorplan.py:124]: you can't have a raw string and a formatted string together

File: build.py [hunk 3]
```diff
@@ -22,6 +130,9 @@ def setup_options(chip):
     cur_dir = os.path.dirname(os.path.realpath(__file__))
     chip.add('option', 'define', f'MEM_ROOT={cur_dir}')
 
+    chip.add('option', 'scpath', os.path.dirname(__file__))
```
[the remark this answers is not in the document]
Comment (gadfort) [reply build.py:133]: If you don't run it from the zerosoc directory, everything fails

File: .flake8 (hunk 1 above)
[the remark this answers is not in the document]
Comment (nmoroze) [reply .flake8:1]: Cool, definitely seems less aggressive/opinionated than black

File: build.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nmoroze) [reply build.py:133]: Ah yeah, makes sense. Might be worth a comment

[the remark this answers is not in the document]
Comment (gadfort) [reply build.py:135]: added

**Review by nmoroze — APPROVED** [review]

**Pull request merged**